### PR TITLE
facedetector now returns a list of tensors containing the boxes x image

### PR DESCRIPTION
**Pull request by lferraz**
#### Changes
Now the FaceDetector returns lists of tensors instead of a tensor. It returns a list because not in all the images the number of boxes is the same.

Fixes # (issue)
When passing batches to the face detector the code failed because was partially not ready to support them.

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by edgarriba — COMMENTED** [review]
can you add a quick test to validate i/o shapes?

**Comment by lferraz** [comment]
> can you add a quick test to validate i/o shapes?

I improved one existing tests adding batches and some shape checks but it is not as simple as other ops.

File: kornia/contrib/face_detection.py [hunk 1]
```diff
@@ -181,32 +181,43 @@ def postprocess(self, data: Dict[str, torch.Tensor], height: int, width: int) ->
...
+            # keep top-K faster NMS
+            batched_dets.append(dets[: self.keep_top_k])
+        return batched_dets
 
-    def forward(self, image: torch.Tensor) -> torch.Tensor:
+    def forward(self, image: torch.Tensor) -> List[torch.Tensor]:
+        r"""Detect faces in a given batch of images.
+
+        Args:
+            image (torch.Tensor): batch of images :math:`(B,3,H,W)`
```
Comment (edgarriba) [inline kornia/contrib/face_detection.py:216]:
```suggestion
            image: batch of images :math:`(B,3,H,W)`.
```

File: kornia/contrib/face_detection.py [hunk 2]
```diff
@@ -181,32 +181,43 @@ def postprocess(self, data: Dict[str, torch.Tensor], height: int, width: int) ->
...
+
+        Return:
+            List[torch.Tensor]: list with the boxes found on each image. :math:`Bx(N,15)`
```
Comment (edgarriba) [inline kornia/contrib/face_detection.py:219]:
```suggestion
            List[torch.Tensor]: list with the boxes found on each image. :math:`Bx(N,15)`.
```

File: kornia/contrib/face_detection.py [hunk 3]
```diff
@@ -314,9 +325,11 @@ def forward(self, x: torch.Tensor) -> Dict[str, torch.Tensor]:
         loc_data, conf_data, iou_data = head_data.split((14, 2, 1), dim=-1)
 
         if self.phase == "test":
-            loc_data = loc_data.view(-1, 14)
-            conf_data = torch.softmax(conf_data.view(-1, self.num_classes), dim=-1)
-            iou_data = iou_data.view(-1, 1)
+            # trick to make it work with batches
+            # loc_data = loc_data.view(-1, 14)
```
Comment (edgarriba) [inline kornia/contrib/face_detection.py:329]: remove commented code

File: test/test_contrib.py [hunk 4]
```diff
@@ -576,12 +576,17 @@ class TestFaceDetection:
     def test_smoke(self, device, dtype):
         assert kornia.contrib.FaceDetector().to(device, dtype) is not None
 
-    def test_valid(self, device, dtype):
+    @pytest.mark.parametrize("batch_size", [1, 2, 4])
+    def test_valid(self, batch_size, device, dtype):
         torch.manual_seed(44)
-        img = torch.rand(1, 3, 320, 320, device=device, dtype=dtype)
+        img = torch.rand(3, 320, 320, device=device, dtype=dtype)
```
Comment (edgarriba) [inline test/test_contrib.py:582]:
```suggestion
        img = torch.rand(batch_size, 3, 320, 320, device=device, dtype=dtype)
```
optional; more clean than stacking

**Review by edgarriba — APPROVED** [review]
LGTM. Minor changes

File: test/test_contrib.py (hunk 4 above)
Comment (lferraz) [reply test/test_contrib.py:582]: totally agree, I did a mess XD

**Pull request merged**